crypto/asn1/asn_mime.c

```diff
@@ -663,7 +663,8 @@ static STACK_OF(MIME_HEADER) *mime_parse_hdr(BIO *bio)
                     mime_debug("Found End Value\n");
                     *p = 0;
                     mhdr = mime_hdr_new(ntmp, strip_ends(q));
-                    sk_MIME_HEADER_push(headers, mhdr);
+                    if (!sk_MIME_HEADER_push(headers, mhdr))
+                        goto err;
                     ntmp = NULL;
```

Comment:
Probably we should set `mhdr` to NULL here after a successful push. If something else called `goto err` then `mhdr` would be freed, and `headers` would be pop_freed == double free. Actually that can't happen in the current code because all instances of `goto err` have `mhdr` set immediately before it - but I could easily envisage new `goto err` calls being added in the future.